tests/test_sync_manager.py

```diff
@@ -172,26 +197,71 @@ def define_dataset1(self, dataset):
         # dataset.data.user_id = 'user1'
         return dataset
 
+    def define_datastore2(self, datastore):
+        datastore.data.concurrency = 1
+        datastore.data.engine_name = 'no_op_engine'
+        return datastore
+
+    def define_workflow2(self, workflow):
+        workflow.data.concurrency = 1
+        workflow.data.engine_name = 'no_op_engine'
+        return workflow
+
+    def define_action3(self, action):
```

Comment:
Can you define an action with args to test the earlier case that I mentioned about the args dictionary getting overwritten?